src/popup/router/components/RadioButton.vue

```diff
@@ -0,0 +1,91 @@
+<template>
+  <label
+    class="radio-container"
+    :class="{ disabled }"
+  >
+    <input
+      :disabled="disabled"
+      :value="value"
+      :type="type"
+      :name="name"
+      @change="handleChange"
+    >
+    <span
+      class="radio-dot"
+      :class="{ checked: value }"
+    />
+    <slot class="radio-holder" />
+  </label>
+</template>
+
+<script>
+export default {
+  props: {
+    value: { type: [String, Number, Boolean], default: '' },
+    type: { type: String, default: 'checkbox' },
+    name: { type: String, default: '' },
+    disabled: { type: Boolean },
+  },
+  methods: {
+    handleChange(event) {
+      if (this.disabled) return;
+      this.$emit('input', event.target.checked);
```

Comment:
I suggest following:
```
if (!this.disabled) {
  this.$emit(...
}
```

Please bare in mind that it is a good practice to decide if function should return something or not.